src/plots/cartesian/layout_attributes.js

```diff
@@ -1003,6 +1003,18 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    shift: {
+        valType: 'any',
+        dflt: false,
```

Comment:
What happens / should happen if one sets `shift` to zero?
Can't we use zero as default instead of *false*?

Reply:
`shift=0` will cause the axis to directly overlap the one that is anchored to the graph on the same size, which is the same as the default `shift=false`. I do think `shift=false` is more intuitive as the default as it's likely that most people will use `shift=true` -- the numeric values are for rarer cases with really fine-tuned positioning needs.

Reply:
Numbers and `“auto”` (rather than `true`) is a pattern we’ve used several other places

Reply:
If the default off is "false", would it be clearer for the 'on' to be "true"? But happy to switch to "auto" if that's better aligned with the existing standard.

Reply:
I think we should possibly make this attribute an object then provide various current/future options inside it e.g.
```js
shift: {
  enabled: boolean,
  value: number
}
 ```

Reply:
@alexcjohnson @archmoj so we've settled on an API with two params: 

1. `shiftauto`: boolean where you turn on/off the automatic repositioning of axes that would be overlapping. Default = 'false'
2. `shiftval`: numeric where you indicate the number of pixels that you want to move the axis from where it would otherwise be. Default = `0` if `shiftauto=false` and `3` if `shiftauto=true`. 

Is this correct?

Reply:
(already posted this on slack, but repeating so it's visible to all)
OK after discussing further, there's a nice reason to make these separate attributes: so when the axis is auto-shifted, you can use the shift parameter to control the extra spacing between subsequent axes.

- `autoshift`: boolean, default `false`
- `shift`: number, no `dflt` defined in the attribute itself, but the dynamic (and described in the attribute description) default is `0` when `autoshift===false` and `3` (yes +/-3 I guess, depending on the side) if `autoshift===true`